### DISCUSSION TITLE
Extract script from dmp file

### DISCUSSION THREAD

**Discussion by MrJoki007** [tasmota/decode-config#68] [Q&A] [answered]
Hello,
i have a backup of my tasmota, that was exported from the gui (dmp file)
When i try to load the file to the device, it does not come online again.
Now the question:
How can i get the script, that was on the original device? 
console -> edit script

**Comment by curzon01** [comment, marked as answer]
## Linux/Mac

use

`./decode-config.py -s using_script.dmp -g Rules | jq ".script"`

to extracts the script from the stored dmp file `using_script.dmp`

e. g. `./decode-config.py -s using_script.dmp -g Rules | jq ".script" >myscript.txt`

writes the script into a text file `myscript.txt`

## Windows

Because I'm currently have no idea to extract a json key from a command line program (not Powershell) simply use the JSON output of

`decode-config.py -s using_script.dmp -g Rules`

and extract  the text manually from the key "script".

**Reply by MrJoki007** [reply]
Thank you for your answere.
I think my backup did not work.
